packages/gateway-edge-nextjs/src/cal/cal.ts

```diff
@@ -0,0 +1,155 @@
+import { type NextApiRequest, type NextApiResponse } from "next"
+import ical, { type ICalEventData } from "ical-generator"
+import { type Event } from "@dotkomonline/types"
+import { createServerSideHelpers } from "@trpc/react-query/server"
+import { appRouter, createContextInner, transformer } from "@dotkomonline/gateway-trpc"
+import { getServerSession } from "next-auth"
+import jwt from "jsonwebtoken"
+import { authOptions } from "../../../auth/src/web.app"
+
+const helpers = createServerSideHelpers({
+  router: appRouter,
+  ctx: await createContextInner({
+    auth: null,
+  }),
+  transformer, // optional - adds superjson serialization
+})
+
+function eventUrl(req: NextApiRequest, event: Pick<Event, "id">) {
+  const proto = req.headers["x-forwarded-proto"] || "http"
```

Comment:
```suggestion
  const proto = req.headers["x-forwarded-proto"] || "https"
```

I find it unlikely that this will be tested locally that often—I'd rather have some kind of tests for this.

Reply:
testing for what?

Reply:
That it returns an usable URI?

Reply:
Nvm, I didn’t think clear enough about how this is deployed